Toolchain.mk

```
@@ -0,0 +1,174 @@
+BIN = $(CURDIR)/bin
+$(BIN):
+	mkdir -p $(BIN)
+
+PATH := $(BIN):$(PATH)
+
+M = $(shell printf "\033[34;1m▶\033[0m")
+
+CHAINSAW_VERSION = v0.2.10
+CONTROLLER_GEN_VERSION = v0.16.3
+CRDOC_VERSION = v0.6.4
+ENVTEST_VERSION = v0.20.4
+GOLANGCI_LINT_VERSION = v2.1.6
+HELM_DOCS_VERSION = v1.11.0
+HELM_VERSION = v3.16.2
+KIND_VERSION = v0.27.0
+KO_VERSION = v0.16.0
+KUSTOMIZE_VERSION = v5.6.0
+OPERATOR_SDK_VERSION = v1.32.0
+OPM_VERSION = v1.23.2
+YQ_VERSION = v4.35.2
+
+CHAINSAW := $(BIN)/chainsaw-$(CHAINSAW_VERSION)
+$(CHAINSAW): $(BIN)
```

We could get rid of the `ifeq` checks on all the Toolchain targets by telling make that the `$(BIN)` dependency is an `order-only-prerequisite` 
https://www.gnu.org/savannah-checkouts/gnu/make/manual/html_node/Prerequisite-Types.html
```suggestion
$(CHAINSAW): | $(BIN)
```

---

I only just found it when I was trying to figure out why Make kept running the `$(BIN):` target 😅